uktrade/data-hub-frontend
# Migrate to Storybook docs addon (Part 1 of 3)

## Description of change
**Note** - In order to make this work easier to review I have decided to split it into three seperate PRs.

The storybook-readme package has not been updated since 2020 and there have been several deprecation warnings and security alerts related to it. In order to replace it I have introduced Storybook's official addon for this purpose (@storybook/addon-docs).

Instead of installing the addon package directly, I have used the @storybook/addon-essentials package which contains the docs addon and also includes the action and control addons (as well as a few other things).

Annoyingly the addon doesn't work out of the box for multiInstance components due to the way they are exported, however I've come up with a workaround.

This PR does the following:
- Sets up `@storybook/addon-docs` via `@storybook/addon-essentials`
- Removes two addons that come with the essentials package
- Uninstalls the deprecated `storybook-readme` dependency and removes config related to it and other components removed from Storybook.
- Setup all components that didn't previously have readme's to display their proptypes.
- Migrate components from `ArchivePanel` to `FlashMessages` to use the DocsPage. Most of the 

## Test instructions

Run npm install and npm run storybook. Go to any component and look at the 'Docs' page (components named in commits below will have documentation). You should be able to see the docs page.

## Screenshots
### Before

<img width="1468" alt="Screenshot 2022-07-29 at 16 54 02" src="https://user-images.githubusercontent.com/36161814/182883910-57f78b66-0e53-47b8-84fa-ad1b61c2e94e.png">


### After

<img width="1454" alt="Screenshot 2022-07-29 at 16 54 10" src="https://user-images.githubusercontent.com/36161814/182883879-0e173d77-ad15-445e-94e9-32961972e712.png">


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
